test/built-ins/Proxy/setPrototypeOf/internals-call-order.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-proxy-object-internal-methods-and-internal-slots-setprototypeof-v
+description: >
+  Calls target.[[GetPrototypeOf]] after trap result as false and not extensible
+  target
+info: |
+  [[SetPrototypeOf]] (V)
+
+  8. Let booleanTrapResult be ToBoolean(? Call(trap, handler, « target, V »)).
+  9. If booleanTrapResult is false, return false.
+  10. Let extensibleTarget be ? IsExtensible(target).
+  11. If extensibleTarget is true, return true.
+  12. Let targetProto be ? target.[[GetPrototypeOf]]().
+features: [Reflect.setPrototypeOf]
+---*/
+
+var calls = [];
+var proto = {};
+
+var target = new Proxy(Object.create(proto), {
+  isExtensible: function() {
+    calls.push("target.[[IsExtensible]]");
+    return false;
+  },
+  getPrototypeOf: function() {
+    calls.push("target.[[GetPrototypeOf]]");
+    return proto;
+  }
+});
+
+// Proxy must report same extensiblitity as target
+Object.preventExtensions(target);
+
+var proxy = new Proxy(target, {
+  setPrototypeOf: function() {
+    calls.push("proxy.[[setPrototypeOf]]");
+    return true;
+  }
+});
+
+Reflect.setPrototypeOf(proxy, proto);
```

Comment:
Now I remember why I used Reflect instead of Object. The assert.throws using Object.setPrototypeOf might catch a false positive when the Proxy internals throws. See line 35 as necessary configuration step for the assertions which would have caused a TypeError if absent.

In anyway, I need to include an assertion for the result here.